sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -0,0 +1,410 @@
+     * section 4.1.3</a> for more information.
+     *
+     */
+    Azure::Nullable<Models::AttestationSigner> Key;
+
+    /**
+     * The "cty" header property of the JWS.
+     *
+     * See <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.10'>RFC 7515
+     * section 4.1.10</a> for more information.
+     *
+     */
+    Azure::Nullable<std::string> ContentType;
+
+    /**
+     * A URI which can be used to retrieve a JSON Web Key which can verify the signature on
+     * this token.
+     *
+     * See <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.5'>RFC 7515
+     * section 4.1.5</a> for more information.
+     *
+     */
+    Azure::Nullable<std::string> KeyURL;
+
+    /**
+     * Returns the "crit" header property from the JSON Web Signature object.
+     *
+     * See <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.11'>RFC 7515
+     * section 4.1.11</a> for more information.
+     *
+     */
+    Azure::Nullable<std::vector<std::string>> Critical;
+
+    /**
+     * Returns a URI which can be used to retrieve an X.509 certificate which can verify the
+     * signature on this token.
+     *
+     * See <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.5'>RFC 7515
+     * section 4.1.5</a> for more information.
+     *
+     */
+    Azure::Nullable<std::string> X509Url;
+
+    /**
+     * Returns the "typ" header property from the JWS.
+     *
+     * See <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.9'>RFC 7515
+     * section 4.1.9</a> for more information.
+     *
+     */
+    Azure::Nullable<std::string> Type;
+
+    /**
+     * Returns the SHA-1 thumbprint of the leaf certificate in the getCertificateChain.
+     *
+     * See <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.7'>RFC 7515
+     * section 4.1.7</a> for more information.
+     *
+     */
+    Azure::Nullable<std::string> CertificateThumbprint;
+
+    /**
+     * Returns the SHA-256 thumbprint of the leaf certificate in the getCertificateChain.
+     *
+     * See <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.8'>RFC 7515
+     * section 4.1.8</a> for more information.
+     *
+     */
+    Azure::Nullable<std::string> CertificateSha256Thumbprint;
+
+    /**
+     * Returns the signing certificate chain as an AttestationSigner.
+     *
+     * See <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.6'>RFC 7515
+     * section 4.1.6</a> for more information.
+     *
+     */
+    Azure::Nullable<std::vector<std::string>> X509CertificateChain;
+  };
+
+  /** An AttestationToken represents an RFC 7519 JSON Web Token returned from the attestation
+   * service with the specialized body type.
+   * <typeparam name="T"></typeparam> The type which represents the body of the attestation token.
+   */
+  template <typename T> class AttestationToken final {
```

Comment:
move to models?

Reply:
It's not a model type by the definition of model types, so why should it be in models?